surefire-providers/surefire-testng/src/main/java/org/apache/maven/surefire/testng/conf/TestNG740Configurator.java

```diff
@@ -37,33 +35,45 @@
  *
  * @since 3.0.0-M6
  */
-public class TestNG740Configurator extends TestNG60Configurator
+public class TestNG740Configurator
+    extends TestNG60Configurator
 {
     @Override
     public void configure( XmlSuite suite, Map<String, String> options )
         throws TestSetFailedException
     {
-        String threadCountAsString = options.get( THREADCOUNT_PROP );
-        int threadCount = threadCountAsString == null ? 1 : parseInt( threadCountAsString );
-        suite.setThreadCount( threadCount );
-
-        String parallel = options.get( PARALLEL_PROP );
-        if ( parallel != null )
+        // if [threadcount] option in unspecified
```

Comment:
This part is not our goal, right?
It should be reverted back; otherwise we may expect a new bug. Why you deleted the call `suite.setThreadCount( threadCount )`?

Reply:
This code is redundant. The superclass implementation already processes the [threadcount] option. The reason I'm acquiring the default value when the option is unspecified is that the superclass implementation assigns a default value of 1, whereas TestNG currently assigns a default value of 5.

Reply:
Handling of the [dataproviderthreadcount] option here was also redundant and has likewise been removed.